File: RuoYi-Vue-FastApi/common/enums.py
"""
业务枚举定义（对应java版ruoyi-common/enums包）
Python枚举范式：带字段枚举用元组成员 + __init__，一比一复刻java写法
"""
from enum import Enum


class UserStatus(Enum):
    """
    用户状态（对应java版UserStatus）
    """
    OK = ("0", "正常")
    DISABLE = ("1", "停用")
    DELETED = ("2", "已删除")

    def __init__(self, code, info):
        self.code = code
        self.info = info


class BusinessType(Enum):
    """
    业务操作类型（对应java版BusinessType）
    code即java @Log注解business_type落库数字（java枚举ordinal 0-9），勿调整对应关系
    注意：python枚举没有java的ordinal()方法，因此显式携带code字段
    """
    OTHER = (0, "其它")
    INSERT = (1, "新增")
    UPDATE = (2, "修改")
    DELETE = (3, "删除")
    GRANT = (4, "授权")
    EXPORT = (5, "导出")
    IMPORT = (6, "导入")
    FORCE = (7, "强退")
    GENCODE = (8, "生成代码")
    CLEAN = (9, "清空数据")

    def __init__(self, code, info):
        self.code = code
        self.info = info


class OperatorType(Enum):
    """
    操作人类别（对应java版OperatorType；java中OTHER=0, MANAGE=1, MOBILE=2，用显式value对齐）
    """
    OTHER = (0, "其它")
    MANAGE = (1, "后台用户")
    MOBILE = (2, "手机端用户")

    def __init__(self, code, info):
        self.code = code
        self.info = info


class DataSourceType(Enum):
    """
    数据源（对应java版DataSource）
    """
    MASTER = ("master", "主库")
    SLAVE = ("slave", "从库")

    def __init__(self, name_, info):
        self.name_ = name_
        self.info = info


class HttpMethod(Enum):
    """
    http方法（对应java版HttpMethod）
    """
    GET = ("GET", "查")
    POST = ("POST", "增")
    PUT = ("PUT", "改")
    DELETE = ("DELETE", "删")

    def __init__(self, method, action):
        self.method = method
        self.action = action

File: RuoYi-Vue-GO/internal/common/enums/enums.go
// Package enums 通用枚举（对位 Java com.ruoyi.common.enums）。
//
// 约定：全部用 typed const 显式赋值，不用 iota 制造隐式序数；
// 落库值（如 @Log business_type、操作类别、操作状态）必须显式等于 Java 枚举的 ordinal。
// 未移植：DataSourceType（GO 版单数据源，不复刻动态数据源切换）、
// DesensitizedType（脱敏工具枚举，待有模块用到时再补）。
package enums

// BusinessType 业务操作类型（对位 BusinessType.java，@Log business_type 落库值 = 序数 0-9）
type BusinessType int

const (
	BusinessTypeOther   BusinessType = 0 // 其它
	BusinessTypeInsert  BusinessType = 1 // 新增
	BusinessTypeUpdate  BusinessType = 2 // 修改
	BusinessTypeDelete  BusinessType = 3 // 删除
	BusinessTypeGrant   BusinessType = 4 // 授权
	BusinessTypeExport  BusinessType = 5 // 导出
	BusinessTypeImport  BusinessType = 6 // 导入
	BusinessTypeForce   BusinessType = 7 // 强退
	BusinessTypeGencode BusinessType = 8 // 生成代码
	BusinessTypeClean   BusinessType = 9 // 清空数据
)

// String 返回 Java 枚举名，便于日志与对表
func (t BusinessType) String() string {
	switch t {
	case BusinessTypeOther:
		return "OTHER"
	case BusinessTypeInsert:
		return "INSERT"
	case BusinessTypeUpdate:
		return "UPDATE"
	case BusinessTypeDelete:
		return "DELETE"
	case BusinessTypeGrant:
		return "GRANT"
	case BusinessTypeExport:
		return "EXPORT"
	case BusinessTypeImport:
		return "IMPORT"
	case BusinessTypeForce:
		return "FORCE"
	case BusinessTypeGencode:
		return "GENCODE"
	case BusinessTypeClean:
		return "CLEAN"
	default:
		return "UNKNOWN"
	}
}

// OperatorType 操作人类别（对位 OperatorType.java，操作日志 operate_type 落库值 = 序数 0-2）
type OperatorType int

const (
	OperatorTypeOther  OperatorType = 0 // 其它
	OperatorTypeManage OperatorType = 1 // 后台用户
	OperatorTypeMobile OperatorType = 2 // 手机端用户
)

// String 返回 Java 枚举名
func (t OperatorType) String() string {
	switch t {
	case OperatorTypeOther:
		return "OTHER"
	case OperatorTypeManage:
		return "MANAGE"
	case OperatorTypeMobile:
		return "MOBILE"
	default:
		return "UNKNOWN"
	}
}

// BusinessStatus 操作状态（对位 BusinessStatus.java，操作日志 status 落库值 = 序数 0-1）
type BusinessStatus int

const (
	BusinessStatusSuccess BusinessStatus = 0 // 成功
	BusinessStatusFail    BusinessStatus = 1 // 失败
)

// String 返回 Java 枚举名
func (s BusinessStatus) String() string {
	switch s {
	case BusinessStatusSuccess:
		return "SUCCESS"
	case BusinessStatusFail:
		return "FAIL"
	default:
		return "UNKNOWN"
	}
}

// LimitType 限流类型（对位 LimitType.java）
type LimitType int

const (
	LimitTypeDefault LimitType = 0 // 默认策略全局限流
	LimitTypeIP      LimitType = 1 // 根据请求者 IP 进行限流
)

// String 返回 Java 枚举名
func (t LimitType) String() string {
	switch t {
	case LimitTypeDefault:
		return "DEFAULT"
	case LimitTypeIP:
		return "IP"
	default:
		return "UNKNOWN"
	}
}
